src/compute-client/src/protocol/command.proto

```diff
@@ -59,6 +63,9 @@ message ProtoPeek {
     mz_expr.relation.ProtoRowSetFinishing finishing = 5;
     mz_expr.linear.ProtoSafeMfpPlan map_filter_project = 6;
     map<string, string> otel_ctx = 7;
+    oneof target {
```

Comment:
I've done almost all of your suggestion... except renaming the id field, since I'd like to minimize thrashing if I have to unwind this stuff. Left a comment that it should be cleaned up in the future, though!